metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -223,6 +225,17 @@ private boolean ensureDbInit() {
       partColumnQuery = pm.newQuery(MPartitionColumnStatistics.class, "dbName == ''");
       partColumnQuery.execute();
 
+      /*
+        these queries for the notification related tables have to be executed so
+        that the tables are created. This was not required earlier because we were
+        interacting with these tables via DataNucleus so it would create them if
+        they did not exist (mostly used in test, schematool should be used for production).
+        however this has been changed and we used direct SQL
+        queries via DataNucleus to interact with them now.
+       */
+      pm.newQuery(MNotificationLog.class, "dbName == ''").execute();
+      pm.newQuery(MNotificationNextId.class, "nextEventId < -1").execute();
```

Comment:
As I understand, if direct sql fails, then the SQL query falls back to datanucleus. Also, we're using datanucleus when we add a notification event. Isn't this table initialized automatically when we fall back or use datanucleus?

Are we using direct-sql for events as well? I don't find this code.

Reply:
yeh i think we are not using directSQL from MNotificationLog as of now, just thought of adding both the dependent tables here for creation though